coderd/database/modelmethods.go

```diff
@@ -148,6 +148,10 @@ func (g Group) RBACObject() rbac.Object {
 		InOrg(g.OrganizationID)
 }
 
+func (w GetWorkspaceByAgentIDRow) RBACObject() rbac.Object {
```

Comment:
This is essentially expected behaviour, but it's just annoying because of how our current authorization matches up with our database structure.

If we want to authorize GetWorkspaceByAgentID we need to know what sort of resource to authorize against. We want only folks who can read the workspace to be able to read a GetWorkspaceByAgentIDRow for that workspace.

If you look at the changes to the GetWorkspaceByAgentID query in `queries/workspaces.sql`, you can see that we are embedding the workspace in the returned rows.

Reply:
Thanks for clarification 👍

Reply:
Yea. It's a consequence too of the fact sqlc generates a new model type for each query if you are not selecting on a table. Ideally I could control the model type, make `Workspace` embedded anonymously, and just attach the `TemplateName` as an extra field.

Then it'd inherit the authz properties of the embedded workspace without having to write this function.